source/extensions/filters/http/oauth2/oauth_client.cc

```diff
@@ -37,20 +37,21 @@ void OAuth2ClientImpl::asyncGetAccessToken(const std::string& auth_code,
   const auto encoded_secret = Http::Utility::PercentEncoding::encode(secret, ":/=&?");
   const auto encoded_cb_url = Http::Utility::PercentEncoding::encode(cb_url, ":/=&?");
 
+  const auto basic_auth_token = absl::StrCat(client_id, ":", secret);
+  const auto encoded_token = Base64::encode(basic_auth_token.data(), basic_auth_token.size());
+  const auto basic_auth_header_value = absl::StrCat("Basic ", encoded_token);
+
```

Comment:
Ah for some reason I thought we were pre-encoding the data here during filter creation. The way it is now this probably should be inside the switch case. Similarly, I think values that are only used by the UrlEncodedBody should be inside that case block, leaving only shared ones (encoded_cb_url, secret) on top. Sorry for the back and forth here!

Reply:
@snowp updated the code. `encoded_cb_url` is common for both the cases